books/views.py

```diff
@@ -195,3 +197,38 @@ def create_book_holder(self, book, holder, status):
             status=status,
         )
         book.save()
+
+
+class BookInterest(BaseLoanView):
+    model = Book
+    success_url = "email_success"
+    template_name_suffix = "_confirm_interest_item"
+    new_status = "OL"
+
+    # def get_new_holder(self, request, *args, **kwargs):
+    #     return request.user
+
+    def create_book_holder(self, book, holder, status):
+        # add entry in Holder table?
+        # probably better to create some other log
+
+        current_user = self.request.user
+        from_email = current_user.email
+        to_email = [holder.email]
+        subject = "Bookx: " + current_user.username + " and " + book.title
+        message = (
+            "Our records show that you have borrowed the book "
+            + book.title
+            + ". "
+            + current_user.username
+            + " is interested in this book. Please could you let them know "
+            + "when you will return the book."
+        )
```

Comment:
You can use regular template rendering for composing email text as well, instead of having these in your view.

It's not a very good place for sending emails anyway, it would be nicer to move this business logic elsewhere (ex. email_notifications.py ?). It is likely these will need to be converted to async tasks, as they are blocking operations (that may also fail for random connectivity issues).

Reply:
Also, on a practical note, you do not need the "+" to link strings that are already within brackets `()`. You can also take advantage of `f` strings to put variables in there. Generally adding variables to strings with `+` is not encouraged

So you can have things like:

```
subject = f"Bookx: {current_user.username} and {book.title}"
message = (
    f"Our records  show that you have borrowed the book {book.title}. "
    f"{current_user.username} is interested in this book. "
    "Please could you let them know when you will return the book. "
)
```

`f` strings are also very efficient. Alternatively you can use string.format() or `%` style formatting. Just avoid `+` 😂